### PR TITLE
Add warning popup to order cycle list [OFN-12775]

### LINKED ISSUE #12775: [Admin] [Order Cycles] Add warning popup to order cycle list as well
## Description

Follow up of #12653 

The same popup should be shown too when updating the start or end date of an order cycle, which has orders attached to it, on the order cycle list, page `/admin/order_cycles`.

## Expected Behavior

The popup from #12653 should be shown under the same conditions.

## Actual Behaviour

There is no popup.

## Steps to Reproduce

1. Have an order cycle with an order.
2. Go to the order cycle list page.
3. Change the start or end date/time of that order cycle and click 'Save changes'.
4. Observe that there is no popup.

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used:
* Browser name and version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by wandji20**
#### What? Why?

- Closes #12775

Include OC warning modal to OC list page

#### What should we test?
- Changing a datetime value from OC list should trigger the confirmation modal

- Visit ... page.
localhost:3000/admin/order_cycles

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [x] User facing changes
- [ ] Technical changes only

Include OC warning modal to OC list page

#### Dependencies
N/A

#### Documentation updates
N/A

### CODE REVIEW

File: spec/controllers/admin/order_cycles_controller_spec.rb [hunk 1]
```diff
@@ -323,13 +328,18 @@ def build_resource
...
+      let(:form_mock) {
+        instance_double(OrderCycles::FormService, save: true)
+      }
 
       before { allow(controller).to receive(:spree_current_user) { user } }
 
       context "as a manager of the coordinator" do
         let(:user) { coordinator.owner }
-        let(:expected) { [order_cycle, allowed.merge(restricted), user] }
+        let(:error_class) { Admin::OrderCyclesController::DateTimeChangeError }
```
Comment (rioug) [inline spec/controllers/admin/order_cycles_controller_spec.rb:339]: I believe this is not used ?

File: app/services/sets/order_cycle_set.rb [hunk 2]
```diff
@@ -3,7 +3,22 @@
...
+
       super(OrderCycle, collection, attributes)
     end
+
+    def process(found_element, attributes)
+      if @confirm_datetime_change && found_element.orders.exists? &&
+         ((attributes.key?(:orders_open_at) &&
+           !found_element.same_datetime_value(:orders_open_at, attributes[:orders_open_at])) ||
+          (attributes.key?(:orders_close_at) &&
+            !found_element.same_datetime_value(:orders_close_at, attributes[:orders_close_at])))
```
Comment (rioug) [inline app/services/sets/order_cycle_set.rb:17]: This is very hard to read, is there any chance to simplify this ? Maybe split the checks in their own method ?

**Review by rioug — CHANGES_REQUESTED** [review]
Thanks @wandji20 , this looks mostly good but I think we can make the code a bit easier to read. 
I am also wondering if we could extract `same_datetime_value` to an helper ? Anyway Its not necessary for now.

File: spec/controllers/admin/order_cycles_controller_spec.rb (hunk 1 above)
Comment (wandji20) [reply spec/controllers/admin/order_cycles_controller_spec.rb:339]: I needed to add `error_class` to match the mocked `OrderCycles::FormService` class as `@error_class` is now set in the initializer

File: app/services/sets/order_cycle_set.rb (hunk 2 above)
Comment (rioug) [reply app/services/sets/order_cycle_set.rb]: Much better :pray:

**Review by rioug — APPROVED** [review]

**Comment by filipefurtad0** [comment]
Hey @wandji20 ,

Nice! Thanks for another PR.

I could observe that a pop-up is now also appearing on the order cycles list page:

image

This happens if an order (currently _any_ order, please see PS below) is associated with the order cycle. The pop-up works as expected, i.e.:
- accepting keeps changes
- cancelling discards changes

This looks great, merging!!

PS: I think one improvement to this PR would be to consider the state of orders; I would suggest that ideally the pop-up only appears if there are orders in the `complete` or `resumed` states (these are the states which tag orders to be included in reports, customer balances and other relevant features). Currently, the pop-up appears even if customers have not completed the order (for example, orders which are in the `cart` state). Please take this comment as a simple FYI - since you're doing so well with the OFN code, I thought this could be useful info, for future PRs. Again, many thanks for your contributions :muscle:

**Comment by filipefurtad0** [comment]
I've verified the failing specs to be flaky. Merging.

**Pull request merged**